x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -744,15 +767,20 @@ export class TaskRunner<
       });
     }
 
+    const transformStateForTaskState = (
```

Comment:
I am not happy with this solution but this was the most explicit way without refactoring.
As Patrick said, this file is in dire need of refactoring :)